### DISCUSSION TITLE
Support for LM73 Temperature Sensor

### DISCUSSION THREAD

**Discussion by domdfcoding** [arendst/Tasmota#17043] [Ideas]
Would it be possible to add support for the LM73 temperature sensor by Texas Instruments? It uses i2c, and there are six registers, of which only the temperature register is probably going to be useful for Tasmota. 

I found some code [here](https://www.pcbway.com/project/shareproject/Build_your_own_IoT_Controller.html) (scroll down to "code") for communicating with the sensor. It's pretty simple. 

I haven't contributed to Tasmota before, but I'm happy to have a go at writing a driver if someone can point me in the right direction for where the code should go.

Product page and datasheet etc.: https://www.ti.com/product/LM73

**Comment by barbudor** [comment]
You could start from the LM75 driver `xsns_26_lm75ad.ino` which is very similar
You need to assign a new Driver ID. The next free one is 101 so replace XNS_26 by XSNS_101 and driver would be `xdrv_101_lm73.ino`
You need to assign a new I2CDriver ID. The next free one is 73 so replace XI2C_20 by XI2C_73
You will have to adapt some defines, the temperature reading as the LM73 as a smaller LSB than the LM75

Should be very straight forward

Alternatively the most efficient way would be to update the LM75 driver to support autoidentification based on the identification register.
Unfortunately don't have any LM73 in the part bin, only LM75
